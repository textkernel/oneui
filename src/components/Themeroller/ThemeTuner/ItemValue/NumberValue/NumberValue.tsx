import * as React from 'react';
import { Input } from '../../../../Input';
import { ThemeNumberItem } from '../../../../../themes/themerollerConfig';

interface Props {
    item: ThemeNumberItem;
    onChange: (value: ThemeNumberItem) => void;
    ariaRole?: string;
}

export const NumberValue: React.FC<Props> = ({ item, onChange, ariaRole }) => {
    const handleOnChange = (event) => {
        onChange({
            ...item,
            value: event.target.value,
        });
    };
    return (
        <Input
            type="number"
            size="small"
            value={item.value}
            onChange={handleOnChange}
            role={ariaRole}
        />
    );
};

NumberValue.displayName = 'NumberValue';
