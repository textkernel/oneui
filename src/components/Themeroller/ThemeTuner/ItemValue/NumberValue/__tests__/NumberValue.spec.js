import React from 'react';
import toJson from 'enzyme-to-json';
import { NumberValue } from '../NumberValue';

describe('NumberValue component', () => {
    const onChangeMock = jest.fn();
    const item = { value: '100' };

    afterEach(() => {
        jest.resetAllMocks();
    });

    it('should render component correctly', () => {
        const wrapper = mount(<NumberValue item={item} onChange={onChangeMock} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should invoke onChange callback when input is changed', () => {
        const wrapper = mount(<NumberValue item={item} onChange={onChangeMock} />);
        wrapper.find('input').simulate('change', {
            target: {
                value: '200',
            },
        });
        expect(onChangeMock).toHaveBeenCalledWith({ value: '200' });
    });
});
