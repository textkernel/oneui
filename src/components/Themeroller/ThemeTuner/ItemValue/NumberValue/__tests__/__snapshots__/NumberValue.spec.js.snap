// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberValue component should render component correctly 1`] = `
<NumberValue
  item={
    Object {
      "value": "100",
    }
  }
  onChange={[MockFunction]}
>
  <Input
    disabled={false}
    isBlock={false}
    onChange={[Function]}
    size="small"
    type="number"
    value="100"
  >
    <input
      className="Input Input--size_small"
      data-lpignore={true}
      disabled={false}
      onChange={[Function]}
      type="number"
      value="100"
    />
  </Input>
</NumberValue>
`;
